src/substrait/textplan/TypeConversion.cpp

```diff
@@ -0,0 +1,151 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include "substrait/textplan/TypeConversion.h"
+
+#include "substrait/type/Type.h"
+
+namespace io::substrait::textplan {
+
+namespace {
+
+std::string kNullSignature = "?";
+
+} // namespace
+
+// Converts a Type proto directly into its text signature form.
+std::string typeToText(const ::substrait::proto::Type& type) {
```

Comment:
how about refactor this method as  `std::share_ptr<ParameterType> fromSubstraitType(const ::substrait::proto::Type& type)` and move into header `Type.h` , and then we can get the signature via  ParameterType::signature() ?

Reply:
I considered including the methods there but I was concerned about adding the protobufs as a dependency.  If this isn't an issue I'm up for moving this there.  @chaojun-zhang